killerbee/dev_cc253x.py

```diff
@@ -0,0 +1,272 @@
+'''
+CC253x support is contributed by Scytmo.
+'''
+
+# Import USB support depending on version of pyUSB
+try:
+    import usb.core
+    import usb.util
+    import sys
+    print >>sys.stderr, "Warning: You are using pyUSB 1.x, support is in beta."
+except ImportError:
+    import usb
+    print >>sys.stderr, "Error: You are using pyUSB 0.x, not supported for CC253x."
+    sys.exit()
+
+import struct
+import time
+from datetime import datetime
+from kbutils import KBCapabilities
+
+
+class CC253x:
+    USB_DIR_OUT        = 0x40
+    USB_DIR_IN         = 0xC0
+    USB_POWER_ON       = 0xC5
+    USB_POWER_STATUS   = 0xC6
+    USB_XFER_START     = 0xD0
+    USB_XFER_STOP      = 0xD1
+    USB_XFER_CHAN      = 0xD2
+    USB_CC2530_DATA_EP = 0x82
+    USB_CC2531_DATA_EP = 0x83
+
+    VARIANT_CC2530 = 0
+    VARIANT_CC2531 = 1
+
+    def __init__(self, dev, bus, variant):
+        #TODO deprecate bus param, and dev becomes a usb.core.Device object, not a string in pyUSB 1.x use
+        '''
+        Instantiates the KillerBee class for Zigduino running GoodFET firmware.
+        @type dev:   String
+        @param dev:  PyUSB device
+        @return: None
+        @rtype: None
+        '''
+        if variant == CC253x.VARIANT_CC2530:
+            self._data_ep = CC253x.USB_CC2530_DATA_EP
+        else:
+            self._data_ep = CC253x.USB_CC2531_DATA_EP
+        self._channel = None
+        self.dev = dev
+        self.name = "<unknown>"
+
+        self.__stream_open = False
+        self.capabilities = KBCapabilities()
+        self.__set_capabilities()
+
+        # Set default configuration
+        self.dev.set_configuration()
+
+        # get name from USB descriptor
+        self.name = usb.util.get_string(self.dev, self.dev.iProduct)
+
+        # Get wMaxPacketSize from the data endpoint
+        for cfg in self.dev:
+            for intf in cfg:
+                for ep in intf:
+                    if ep.bEndpointAddress == self._data_ep:
+                        self._maxPacketSize = ep.wMaxPacketSize
+
+
+    def close(self):
+        pass
```

Comment:
In my version I was checking to see if the device was in capture mode, and if so halting that.